### PR TITLE
Cover the migration edge cases that risk data loss

### LINKED ISSUE #1050: Data-safety coverage for schema migrations at the database level
Migrations apply real DDL and data operations to production databases, and the current test
coverage leaves gaps at the database level: crash-interrupted applies, concurrent nodes working
on the same database, and operations running over real pre-existing rows.

The goal is to verify, against a real database, that no supported migration scenario can lose
or corrupt data - interrupted work resumes cleanly, concurrent work applies exactly once, and
operations that cannot be applied safely are refused with nothing partially applied.

### PR DESCRIPTION
**Pull request by bartblast**
Closes #1050

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved migration validation for numeric conversions, including overflow, rounding, extreme values, invalid formats, and special floating-point values.
  * Fixed enum remapping and rename handling when ordering changes, while preserving valid carried-over values.
  * Improved migration safety during concurrent application, interruptions, refusals, drift, and index recovery.
  * Fixed backfills and grant preservation across entity and role changes.

* **Tests**
  * Added comprehensive integration coverage for migration atomicity, resumption, concurrency, locking, backfills, drift, and database isolation.

### CODE REVIEW

**Comment by bartblast** [comment]
@coderabbitai pause

File: test/elixir/hologram/migration/lock_contention_test.exs [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+  defp queued_advisory_lock_count(session) do
+    statement = """
+    SELECT COUNT(*) FROM pg_catalog.pg_locks
+    WHERE "locktype" = 'advisory' AND NOT "granted"
+    """
+
+    %{rows: [[count]]} = Postgrex.query!(session, statement, [])
+
+    count
+  end
```
[the remark this answers is not in the document]
Comment (bartblast) [reply test/elixir/hologram/migration/lock_contention_test.exs:153]: The mechanism checks out - `pg_locks` is cluster-wide and advisory locks do carry their database's OID, confirmed against a live server.

The scenario doesn't, though: the `--partitions 8` job is the feature app, and each partition is its own runner that starts its own Postgres, so partitions never share a server. The lib job runs alone with its own. And inside a single run, ExUnit runs `async: false` modules serially, so nothing else in this suite is holding a lock while these assertions read.

Taking it anyway, for a plainer reason: `build_states/1` two functions up already scopes to `current_database()`, and this one should say the same thing it means - "is anyone queued on an advisory lock in THIS database". It also closes the case that is genuinely reachable, which is a second `mix test` process against the same local server.

Re-ran the mutation that binds these assertions (`pg_try_advisory_lock` back to `pg_advisory_lock`) against the scoped query - still fails with `left: 1, right: 0`, so the narrower predicate still catches the queued lock it exists to catch.

File: test/elixir/support/scratch_database_case.ex [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      import Hologram.Test.Helpers
+      import Hologram.Test.ScratchDatabaseCase
+
+      @moduletag :scratch_database
+    end
+  end
+
+  setup do
+    database = "hologram_scratch_#{System.unique_integer([:positive])}"
+    quoted_database = Mapper.quote_identifier(database)
```
[the remark this answers is not in the document]
Comment (bartblast) [reply test/elixir/support/scratch_database_case.ex]: Holds, and it is worse than "can generate the same name" - the values from independent VMs are adjacent, not merely possibly equal. Three fresh ones here answered 1348, 1355 and 1349, and each hands out a run of values from that same band, so two `mix test` runs against one server would collide many times over rather than once by bad luck.

Fixed by putting the OS process id before the counter: `hologram_scratch_<os pid>_<counter>`. That separates concurrent runs on a machine, and unlike a random suffix it says which run owns a database if a killed one leaves one behind - which this tier has a standing note about. The prefix is untouched, so the leftover probe and the harness's own name assertion still hold.
